From 43a36bd447c69cf057115aed947672139e97ce23 Mon Sep 17 00:00:00 2001
From: Henrik Andersson <hean01@users.sourceforge.net>
Date: Fri, 25 Feb 2011 18:13:50 +0100
Subject: [PATCH] Fixes the cmake for binary package build,
 default -march=native is used which is wanted on local builds of
 git source, when building a package -DBINARY_PACKAGE_BUILD=1 should
 be provided to cmake to use -mtune=generic for generic optimization..

---
 CMakeLists.txt     |    1 +
 src/CMakeLists.txt |   10 ++++++++--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c252af6..4050e46 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,7 @@ option(DONT_INSTALL_GCONF_SCHEMAS "Dont install gconf schemas, usefull for packa
 option(BUILD_USERMANUAL "Build all the versions of the usermanual." OFF)
 option(INSTALL_IOP_EXPERIMENTAL "Also install unstable, unfinished, broken, and likely-to-change-soon plugins." OFF)
 option(INSTALL_IOP_LEGACY "Also install old plugins we want to get rid of." OFF)
+option(BINARY_PACKAGE_BUILD "Sets march optimization to generic" OFF)
 
 #
 # Set platform defaults...
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1c01f32..da68c3a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -208,10 +208,16 @@ if(NOT WIN32)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
 endif(NOT WIN32)
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2 -mfpmath=sse -march=generic -g")
+if(NOT BINARY_PACKAGE_BUILD)
+	set(MARCH "-march=native")	
+else()
+	set(MARCH "-mtune=generic")
+endif()
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2 -mfpmath=sse ${MARCH} -g")
 set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -ffast-math -fexpensive-optimizations")
 set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -g -D_DEBUG")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2 -mfpmath=sse -march=generic -g")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2 -mfpmath=sse ${MARCH} -g")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -ffast-math -fexpensive-optimizations")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -g -D_DEBUG")
   
-- 
1.7.0.1
